modules/core/src/main/java/org/apache/ignite/internal/processors/rest/client/message/GridClientAuthenticationRequest.java

```diff
@@ -50,13 +51,23 @@ public void credentials(Object cred) {
     @Override public void writeExternal(ObjectOutput out) throws IOException {
         super.writeExternal(out);
 
+        U.writeString(out, login());
+        U.writeString(out, password());
```

Comment:
Login and password already included into `cred` object, we don't need to pass it twice.
Also login and password fields are useless and should be removed at all from `GridClientAbstractMessage`
`userAttrs` field should be moved to `GridClientAuthenticationRequest`

Reply:
fixed